### PR TITLE
Refs Task #397767 #397779 payload history list, search, show

### PR DESCRIPTION
**Pull request by nguyenthinhhn**
## Related Tickets

- [#397767](https://dev.sun-asterisk.com/issues/397767)
- [#397779](https://dev.sun-asterisk.com/issues/397779)
- [#397797](https://dev.sun-asterisk.com/issues/397797)

## What's this PR do ?

- [x] Payload history list
- [x] Payload history  search
- [x] Payload history  detail

## Performance

- [ ] Resolved n + 1 query
- [ ] Run explain query already
- [ ] Time run rake task : 1000 ms

## Checklist

- [x] It was tested in local success?
- [x] Fill link PR into ticket and the opposite
- [x] Note purpose/reason, solution, scope of influence into ticket

## ENV note

```
```

## Notes
Screenshot from 2019-12-17 11-14-19
Screenshot from 2019-12-17 13-03-43

### CODE REVIEW

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>failure</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr></table>

File: app/Repositories/Eloquents/MessageHistoryRepository.php [hunk 1]
```diff
@@ -24,4 +24,17 @@ public function getAllAndSearch($id, $keyword)
             return $query->paginate($perPage);
         }
     }
+
+    public function AdminGetAllAndSearch($id, $keyword)
```
Comment (framgiaciv3) [inline app/Repositories/Eloquents/MessageHistoryRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR1.Methods.CamelCapsMethodName.NotCamelCaps</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/MessageHistoryRepository.php</br>**Line**: 28</br>Method name "MessageHistoryRepository::AdminGetAllAndSearch" is not in camel caps format

File: app/Http/Controllers/Admin/PayloadHistoryController.php [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+use Illuminate\Http\Request;
+use App\Http\Controllers\Controller;
+use App\Models\PayloadHistory;
+use App\Enums\PayloadHistoryStatus;
+use App\Repositories\Interfaces\WebhookRepositoryInterface as WebhookRepository;
+use App\Repositories\Interfaces\MessageHistoryRepositoryInterface as MessageHistoryRepository;
+use App\Repositories\Interfaces\PayloadHistoryRepositoryInterface as PayloadHistoryRepository;
+
+class PayloadHistoryController extends Controller
```
Comment (framgiaciv3) [inline app/Http/Controllers/Admin/PayloadHistoryController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.WhiteSpace.DisallowTabIndent.TabsUsed</br>**Severity**: error</br>**File**: app/Http/Controllers/Admin/PayloadHistoryController.php</br>**Line**: 15</br>Spaces must be used to indent lines; tabs are not allowed

File: app/Http/Controllers/Admin/PayloadHistoryController.php [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index(Request $request)
+    {
+        $searchParams = $request->search;
+        $perPage = config('paginate.perPage');
+        $payloadHistories = $this->payloadHistoryRepository->getAllAndSearch($perPage, $searchParams);
+        $webhooks = $this->webhookRepository->getAll()->pluck('id', 'name');
+        $payloadHistoryStatuses = PayloadHistoryStatus::toArray();
```
Comment (framgiaciv3) [inline app/Http/Controllers/Admin/PayloadHistoryController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.Files.LineLength.TooLong</br>**Severity**: warning</br>**File**: app/Http/Controllers/Admin/PayloadHistoryController.php</br>**Line**: 42</br>Line exceeds 120 characters; contains 121 characters

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>failure</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr></table>

File: app/Repositories/Eloquents/MessageHistoryRepository.php (hunk 1 above)
Comment (framgiaciv3) [inline app/Repositories/Eloquents/MessageHistoryRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR1.Methods.CamelCapsMethodName.NotCamelCaps</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/MessageHistoryRepository.php</br>**Line**: 28</br>Method name "MessageHistoryRepository::AdminGetAllAndSearch" is not in camel caps format

File: app/Http/Controllers/Admin/PayloadHistoryController.php [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+namespace App\Http\Controllers\Admin;
+
+use Illuminate\Http\Request;
+use App\Http\Controllers\Controller;
+use App\Enums\PayloadHistoryStatus;
+use App\Repositories\Interfaces\WebhookRepositoryInterface as WebhookRepository;
+use App\Repositories\Interfaces\MessageHistoryRepositoryInterface as MessageHistoryRepository;
+use App\Repositories\Interfaces\PayloadHistoryRepositoryInterface as PayloadHistoryRepository;
+
+class PayloadHistoryController extends Controller
```
Comment (framgiaciv3) [inline app/Http/Controllers/Admin/PayloadHistoryController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.WhiteSpace.DisallowTabIndent.TabsUsed</br>**Severity**: error</br>**File**: app/Http/Controllers/Admin/PayloadHistoryController.php</br>**Line**: 14</br>Spaces must be used to indent lines; tabs are not allowed

File: app/Http/Controllers/Admin/PayloadHistoryController.php [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index(Request $request)
+    {
+        $searchParams = $request->search;
+        $perPage = config('paginate.perPage');
+        $payloadHistories = $this->payloadHistoryRepository->getAllAndSearch($perPage, $searchParams);
+        $webhooks = $this->webhookRepository->getAll()->pluck('id', 'name');
+        $payloadHistoryStatuses = PayloadHistoryStatus::toArray();
```
Comment (framgiaciv3) [inline app/Http/Controllers/Admin/PayloadHistoryController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.Files.LineLength.TooLong</br>**Severity**: warning</br>**File**: app/Http/Controllers/Admin/PayloadHistoryController.php</br>**Line**: 41</br>Line exceeds 120 characters; contains 121 characters

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>failure</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr></table>

File: app/Http/Controllers/Admin/PayloadHistoryController.php [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
```
Comment (framgiaciv3) [inline app/Http/Controllers/Admin/PayloadHistoryController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.Functions.FunctionCallArgumentSpacing.NoSpaceAfterComma</br>**Severity**: error</br>**File**: app/Http/Controllers/Admin/PayloadHistoryController.php</br>**Line**: 41</br>No space found after comma in function call

File: app/Http/Controllers/Admin/PayloadHistoryController.php [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+
```
Comment (framgiaciv3) [inline app/Http/Controllers/Admin/PayloadHistoryController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionCallSignature.Indent</br>**Severity**: error</br>**File**: app/Http/Controllers/Admin/PayloadHistoryController.php</br>**Line**: 42</br>Multi-line function call not indented correctly; expected 12 spaces but found 56

File: app/Http/Controllers/Admin/PayloadHistoryController.php [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+        return view('admins.payload_histories.index',compact(
```
Comment (framgiaciv3) [inline app/Http/Controllers/Admin/PayloadHistoryController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionCallSignature.Indent</br>**Severity**: error</br>**File**: app/Http/Controllers/Admin/PayloadHistoryController.php</br>**Line**: 43</br>Multi-line function call not indented correctly; expected 12 spaces but found 56

File: app/Http/Controllers/Admin/PayloadHistoryController.php [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+                                                        'payloadHistories',
```
Comment (framgiaciv3) [inline app/Http/Controllers/Admin/PayloadHistoryController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionCallSignature.Indent</br>**Severity**: error</br>**File**: app/Http/Controllers/Admin/PayloadHistoryController.php</br>**Line**: 44</br>Multi-line function call not indented correctly; expected 12 spaces but found 56

File: app/Http/Controllers/Admin/PayloadHistoryController.php [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+                                                        'webhooks',
```
Comment (framgiaciv3) [inline app/Http/Controllers/Admin/PayloadHistoryController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionCallSignature.Indent</br>**Severity**: error</br>**File**: app/Http/Controllers/Admin/PayloadHistoryController.php</br>**Line**: 45</br>Multi-line function call not indented correctly; expected 8 spaces but found 52

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>failure</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr></table>

File: app/Http/Controllers/Admin/PayloadHistoryController.php [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index(Request $request)
+    {
+        $searchParams = $request->search;
+        $perPage = config('paginate.perPage');
+        $payloadHistories = $this->payloadHistoryRepository->getAllAndSearch($perPage, $searchParams);
+        $webhooks = $this->webhookRepository->getAll()->pluck('id', 'name');
+        $payloadHistoryStatuses = PayloadHistoryStatus::toArray();
```
Comment (framgiaciv3) [inline app/Http/Controllers/Admin/PayloadHistoryController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionCallSignature.ContentAfterOpenBracket</br>**Severity**: error</br>**File**: app/Http/Controllers/Admin/PayloadHistoryController.php</br>**Line**: 41</br>Opening parenthesis of a multi-line function call must be the last content on the line

File: app/Http/Controllers/Admin/PayloadHistoryController.php [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+
```
Comment (framgiaciv3) [inline app/Http/Controllers/Admin/PayloadHistoryController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionCallSignature.Indent</br>**Severity**: error</br>**File**: app/Http/Controllers/Admin/PayloadHistoryController.php</br>**Line**: 42</br>Multi-line function call not indented correctly; expected 8 spaces but found 12

Comment (framgiaciv3) [inline app/Http/Controllers/Admin/PayloadHistoryController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.Functions.FunctionCallArgumentSpacing.NoSpaceAfterComma</br>**Severity**: error</br>**File**: app/Http/Controllers/Admin/PayloadHistoryController.php</br>**Line**: 42</br>No space found after comma in function call

Comment (framgiaciv3) [inline app/Http/Controllers/Admin/PayloadHistoryController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.Functions.FunctionCallArgumentSpacing.NoSpaceAfterComma</br>**Severity**: error</br>**File**: app/Http/Controllers/Admin/PayloadHistoryController.php</br>**Line**: 42</br>No space found after comma in function call

Comment (framgiaciv3) [inline app/Http/Controllers/Admin/PayloadHistoryController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionCallSignature.CloseBracketLine</br>**Severity**: error</br>**File**: app/Http/Controllers/Admin/PayloadHistoryController.php</br>**Line**: 42</br>Closing parenthesis of a multi-line function call must be on a line by itself

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>failure</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr></table>

File: app/Http/Controllers/Admin/PayloadHistoryController.php (hunk 11 above)
Comment (framgiaciv3) [inline app/Http/Controllers/Admin/PayloadHistoryController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionCallSignature.ContentAfterOpenBracket</br>**Severity**: error</br>**File**: app/Http/Controllers/Admin/PayloadHistoryController.php</br>**Line**: 41</br>Opening parenthesis of a multi-line function call must be the last content on the line

File: app/Http/Controllers/Admin/PayloadHistoryController.php [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
```
Comment (framgiaciv3) [inline app/Http/Controllers/Admin/PayloadHistoryController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionCallSignature.Indent</br>**Severity**: error</br>**File**: app/Http/Controllers/Admin/PayloadHistoryController.php</br>**Line**: 42</br>Multi-line function call not indented correctly; expected 8 spaces but found 12

Comment (framgiaciv3) [inline app/Http/Controllers/Admin/PayloadHistoryController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionCallSignature.CloseBracketLine</br>**Severity**: error</br>**File**: app/Http/Controllers/Admin/PayloadHistoryController.php</br>**Line**: 42</br>Closing parenthesis of a multi-line function call must be on a line by itself

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr></table>

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>failure</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr></table>

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>failure</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr></table>

File: app/Http/Controllers/Admin/PayloadHistoryController.php [hunk 14]
```diff
@@ -0,0 +1,57 @@
...
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index(Request $request)
+    {
+        $searchParams = $request->search;
+        $perPage = config('paginate.perPage');
+        $payloadHistories = $this->payloadHistoryRepository->getAllAndSearch($perPage, $searchParams);
+        $webhooks = $this->webhookRepository->getAll()->pluck('id', 'name');
+        $payloadHistoryStatuses = PayloadHistoryStatus::toArray();
```
Comment (framgiaciv3) [inline app/Http/Controllers/Admin/PayloadHistoryController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionCallSignature.ContentAfterOpenBracket</br>**Severity**: error</br>**File**: app/Http/Controllers/Admin/PayloadHistoryController.php</br>**Line**: 41</br>Opening parenthesis of a multi-line function call must be the last content on the line

File: app/Http/Controllers/Admin/PayloadHistoryController.php [hunk 15]
```diff
@@ -0,0 +1,57 @@
...
+
```
Comment (framgiaciv3) [inline app/Http/Controllers/Admin/PayloadHistoryController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionCallSignature.Indent</br>**Severity**: error</br>**File**: app/Http/Controllers/Admin/PayloadHistoryController.php</br>**Line**: 42</br>Multi-line function call not indented correctly; expected 12 spaces but found 20

File: app/Http/Controllers/Admin/PayloadHistoryController.php [hunk 16]
```diff
@@ -0,0 +1,57 @@
...
+        return view('admins.payload_histories.index',
```
Comment (framgiaciv3) [inline app/Http/Controllers/Admin/PayloadHistoryController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionCallSignature.Indent</br>**Severity**: error</br>**File**: app/Http/Controllers/Admin/PayloadHistoryController.php</br>**Line**: 43</br>Multi-line function call not indented correctly; expected 8 spaces but found 16

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr></table>

File: resources/views/admins/payload_histories/show.blade.php [hunk 17]
```diff
@@ -0,0 +1,113 @@
...
+                            'style' => 'display:inline',
+                            'class' => 'form-delete'
+                            ]) }}
+                            {{ Form::button('<i class="fa fa-trash-o"></i> Delete' , [
+                            'type' => 'DELETE',
+                            'class' => 'btn btn-sm btn-danger delete-btn',
+                            'title' => 'Delete'
+                            ]) }}
+                            {{ Form::close() }}
+                        </td>
```
Comment (ghost) [inline resources/views/admins/payload_histories/show.blade.php]: chỗ này bị lệch rồi nè e

Comment (nguyenthinhhn) [reply resources/views/admins/payload_histories/show.blade.php]: ok a

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr></table>

**Review by quanghung97 — APPROVED** [review]
OK

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr></table>

File: app/Repositories/Eloquents/MessageHistoryRepository.php [hunk 18]
```diff
@@ -24,4 +24,17 @@ public function getAllAndSearch($id, $keyword)
             return $query->paginate($perPage);
         }
     }
+
+    public function adminGetAllAndSearch($id, $keyword)
```
Comment (ghost) [inline app/Repositories/Eloquents/MessageHistoryRepository.php]: a ko tháy e dùng $id,

File: resources/views/admins/payload_histories/show.blade.php [hunk 19]
```diff
@@ -0,0 +1,112 @@
+<?php
+use App\Enums\MessageHistoryStatus;
+use App\Enums\PayloadHistoryStatus;
+?>
+
+@extends('layouts.app')
+@section('content')
+<link rel="stylesheet" href="{{ mix('/css/style.css') }}">
+<ul class="breadcrumb breadcrumb-top">
```
Comment (ghost) [inline resources/views/admins/payload_histories/show.blade.php:9]: sửa cái breadcrumb theo format này giúp a với nhé 
image

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr></table>

File: resources/views/admins/payload_histories/show.blade.php [hunk 20]
```diff
@@ -0,0 +1,111 @@
...
+    <li><a href="/admin/history">Payload History</a></li>
```
Comment (ghost) [inline resources/views/admins/payload_histories/show.blade.php]: Payload Histories

File: resources/views/admins/payload_histories/show.blade.php [hunk 21]
```diff
@@ -0,0 +1,111 @@
...
+    <li><a href="">Detail</a></li>
```
Comment (ghost) [inline resources/views/admins/payload_histories/show.blade.php]: bỏ thẻ a đi e

**Review by ghost — APPROVED** [review]

File: app/Repositories/Eloquents/MessageHistoryRepository.php [hunk 22]
```diff
@@ -24,4 +24,17 @@ public function getAllAndSearch($id, $keyword)
             return $query->paginate($perPage);
         }
     }
+
+    public function adminGetAllAndSearch($keyword)
```
Comment (ghost) [inline app/Repositories/Eloquents/MessageHistoryRepository.php]: cái này show all hả e, show theo từng payload_history ms đúng chứ nhỉ

Comment (ghost) [reply app/Repositories/Eloquents/MessageHistoryRepository.php]: 2 hàm này giống nhau mà e nhỉ
https://github.com/sun-asterisk-research/chatwork-forwarder/pull/84/files#diff-77bac4e347b150ee66e0c47cf7e470a5L15

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr></table>

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr></table>

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr></table>

**Review by ghost — APPROVED** [review]

**Comment by phucqd-1305** [comment]
OK

**Pull request merged**